src/codegen.rs

```diff
@@ -627,11 +624,10 @@ batchNormInference((float*) {X}, (float*) {Y}, {N}, {H}, {W}, {C}, (float*) {gam
                         code,
                         hw_map,
                     );
-
-                    let axis = MyAnalysis::get_usize(ids[2], expr);
-
-                    // Pre-ISCA: resnet only does softmax over (1,1000)
-                    assert!(axis == 1, "expected NHWC format");
+                        
+                    // TODO: axis not used since
+                    // softmax c function does softmax over all values
+                    let _axis = MyAnalysis::get_usize(ids[2], expr);
```

Comment:
Hm. The input shape here is `(1, 1000)`, right? As long as the first dimension, batch, is 1, then our softmax kernel works. Can you add an assert here to assert that batch is 1?

Reply:
That is, if we have batch > 1, the kernel breaks (because it will softmax over EVERYTHING rather than splitting the kernel up), but as long as batch == 1, then we're good.

Reply:
is the shape guaranteed to be two dimensions though?

Reply:
Yeah, that should be a fine assumption for now. All the CNNs we're currently working with use softmax as their very last layer, and the input/output shape is `(batch, num_classes)`. You could also assert that the number of dimensions is 2, if you want to be extra safe. That should be the case for all of our current networks.

Reply:
FWIW, my general strategy is that it's okay if we put in asserts that OVER-constrain. Then, at the very worst, you get a false-positive assertion failure, and you can just refine the condition in the assertion. In your current code, I don't think there's anything to stop us from generating C code that just straight up breaks, if the batch size is greater than 1 in the softmax, or if the `axis` param from relay is something out of the norm. Then, the bug would come in the form of an error from the C code and would be harder to track down!